channel/contractbackend.go

```diff
@@ -45,6 +45,15 @@ const (
 // create a TxTimedoutError with additional context.
 var errTxTimedOut = errors.New("")
 
+// SharedExpected Nonce is a map of each expected next nonce of all clients.
+var (
+	SharedExpectedNonces map[ChainID]map[common.Address]uint64
```

Comment:
Any reason those two are separate maps instead of a `map[ChainID]map[common.Address]struct{uint64,*sync.Mutex}` (assuming a per-item mutex is needed)? As far as I can tell those two always work in tandem and you basically never need/use `SharedExpectedNonces[a][b]` without using `SharedNonceMtx[a][b]` (though I could be wrong and different parts of perun already use separate maps for a different reason).

Even then the following could be an option, too, grouping them and removing an extra lookup due to having two outer maps:
```
type PerChainData struct {
    expectedNonces map[common.Address]uint64
    mutexes map[common.Address]*sync.Mutex
}
map[chainID]PerChainData
```

Reply:
Another note: Do those need to be public? Might be easier to reason about the mutex behavior if only this package can access them.